packages/gatsby-plugin-netlify/src/build-headers-program.js

```diff
@@ -26,10 +26,19 @@ function validHeaders(headers) {
     headers,
     (headersList, path) =>
       _.isArray(headersList) &&
-      _.every(
-        headersList,
-        header => _.isString(header) && getHeaderName(header)
-      )
+      _.every(headersList, header => {
+        if (_.isString(header)) {
+          if (!getHeaderName(header)) {
+            reporter.warn(
```

Comment:
Should this have more severe level? I.e. fail build completely to not deploy potentially broken build?